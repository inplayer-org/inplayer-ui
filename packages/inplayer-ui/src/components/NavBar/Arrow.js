// @flow
import React from 'react';
import styled, { css } from 'styled-components';
import { ifProp, switchProp } from 'styled-tools';
import colors from 'config/colors';
import { transparentize } from 'polished';

export const ArrowContainer = styled.div`
  width: 18px;
  margin: 0 1rem 1rem 1rem;
  text-align: center;
  ${switchProp('section', {
    navbar: css`
      align-self: flex-end;

      &:hover i {
        border-color: ${colors.white};
      }
    `,
    accordion: css`
      align-self: flex-start;
    `,
  })};
`;

export const NavbarArrow = styled.i`
  display: inline-block;
  transition: all 0.5s ease;
  cursor: pointer;
  ${switchProp('section', {
    navbar: css`
      padding: 0.225rem;
      border: solid ${transparentize(0.6, colors.white)};
      border-width: 1px 0 0 1px;
      transform: ${ifProp('open', 'rotate(-45deg)', 'rotate(135deg)')};
    `,
    accordion: css`
      border: solid ${colors.midnightBlue};
      border-width: 1px 0 0 1px;
      transform: ${ifProp('open', 'rotate(135deg)', 'rotate(-45deg)')};
      padding: 0.4rem;
    `,
  })};
`;

type Props = {
  open: boolean,
  section: string,
  onClick: any,
};

const Arrow = ({ open, section, onClick }: Props) => (
  <ArrowContainer section={section} onClick={onClick}>
    <NavbarArrow open={open} section={section} />
  </ArrowContainer>
);

export default Arrow;
