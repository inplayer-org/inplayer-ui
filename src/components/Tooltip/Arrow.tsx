import React from 'react';
import styled from 'styled-components';

type ArrowsProps = {
  width: number;
  background?: string;
  border: string;
  placement: string;
};

const Base = styled.div<ArrowsProps>`
  position: absolute;
  ${({ width }) => width && `width: ${width}px;`};
  ${({ width }) => width && `height: ${width}px;`};
  background: ${({ background, theme }) => background || theme.palette.secondary.main};
`;

const Up = styled(Base)`
  transform: translateX(-50%) translateY(50%) rotateZ(45deg);
  bottom: 100%;
  left: 50%;
  ${({ border }) => border && `border-left: 1px solid ${border};`}
  ${({ border }) => border && `border-top: 1px solid ${border};`}
`;

const Down = styled(Base)`
  transform: translateX(-50%) translateY(-50%) rotateZ(45deg);
  top: 100%;
  left: 50%;
  ${({ border }) => border && `border-right: 1px solid ${border}`}
  ${({ border }) => border && `border-bottom: 1px solid ${border}`}
`;

const Left = styled(Base)`
  transform: translateX(50%) translateY(-50%) rotateZ(45deg);
  right: 100%;
  top: 50%;
  ${({ border }) => border && `border-left: 1px solid ${border}`}
  ${({ border }) => border && `border-bottom: 1px solid ${border}`}
`;

const Right = styled(Base)`
  transform: translateX(-50%) translateY(-50%) rotateZ(45deg);
  left: 100%;
  top: 50%;
  ${({ border }) => border && `border-right: 1px solid ${border}`}
  ${({ border }) => border && `border-top: 1px solid ${border}`}
`;

const arrows: Record<string, any> = {
  left: Right,
  top: Down,
  right: Left,
  bottom: Up,
};

type Props = {
  background: string;
  border: string;
  placement: string;
  width: number;
};

const Arrow = ({ background, border, placement, width }: Props) => {
  const Component = arrows[placement] || arrows.top;
  return width > 0 ? <Component background={background} border={border} width={width} /> : null;
};

export default Arrow;
