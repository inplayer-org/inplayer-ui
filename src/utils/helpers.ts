import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
