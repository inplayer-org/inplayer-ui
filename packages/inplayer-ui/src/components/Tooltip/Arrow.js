// @flow
import React from 'react';
import styled from 'styled-components';
import { ifProp, prop } from 'styled-tools';

import { uiColors } from 'utils';

type Props = {
  background: string,
  border: string,
  placement: string,
  width: number,
};

const Base = styled('div')`
  position: absolute;
  width: ${props => props.width}px;
  height: ${props => props.width}px;
  background: ${ifProp('background', prop('background'), uiColors('secondary.main'))};
`;

const Up = styled(Base)`
  transform: translateX(-50%) translateY(50%) rotateZ(45deg);
  bottom: 100%;
  left: 50%;
  border-left: 1px solid ${props => props.border};
  border-top: 1px solid ${props => props.border};
`;

const Down = styled(Base)`
  transform: translateX(-50%) translateY(-50%) rotateZ(45deg);
  top: 100%;
  left: 50%;
  border-right: 1px solid ${props => props.border};
  border-bottom: 1px solid ${props => props.border};
`;

const Left = styled(Base)`
  transform: translateX(50%) translateY(-50%) rotateZ(45deg);
  right: 100%;
  top: 50%;
  border-left: 1px solid ${props => props.border};
  border-bottom: 1px solid ${props => props.border};
`;

const Right = styled(Base)`
  transform: translateX(-50%) translateY(-50%) rotateZ(45deg);
  left: 100%;
  top: 50%;
  border-right: 1px solid ${props => props.border};
  border-top: 1px solid ${props => props.border};
`;

const arrows = {
  left: Right,
  top: Down,
  right: Left,
  bottom: Up,
};

const Arrow = ({ background, border, placement, width }: Props) => {
  const Component = arrows[placement] || arrows.top;
  return width > 0 && <Component background={background} border={border} width={width} />;
};

export default Arrow;
