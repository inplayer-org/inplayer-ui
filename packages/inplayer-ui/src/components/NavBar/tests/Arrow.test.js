import { generateSnapshots } from 'packages/inplayer-ui/src/components/NavBar/tests/helpers';
import Arrow from '../Arrow';

generateSnapshots(Arrow);
