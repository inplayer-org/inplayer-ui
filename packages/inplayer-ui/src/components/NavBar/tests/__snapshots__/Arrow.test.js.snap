// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow renders correctly 1`] = `
.c0 {
  width: 18px;
  margin: 0 1rem 1rem 1rem;
  text-align: center;
}

.c1 {
  display: inline-block;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  cursor: pointer;
}

<div
  className="c0"
>
  <i
    className="c1"
  />
</div>
`;
